createAccountJava/src/main/java/com/brandon/createaccountjava/view/CreateAccountActivity.java

```diff
@@ -0,0 +1,192 @@
+        final TextInputEditText locationField = binding.locationField;
+        final TextView firstError = binding.firstNameError;
+        final TextView lastError = binding.lastNameError;
+        final TextView emailError = binding.emailError;
+        final TextView dateError = binding.dateError;
+        final TextView locationError = binding.locationError;
+
+        onTextChanged(firstNameField, (String s) -> viewModel.firstNameChanged(s));
+        onTextChanged(lastNameField, (String s) -> viewModel.lastNameChanged(s));
+        onTextChanged(emailField, (String s) -> viewModel.emailChanged(s));
+        onTextChanged(dateField, (String s) -> viewModel.dateChanged(s));
+        onTextChanged(locationField, (String s) -> viewModel.locationChanged(s));
+
+        dateField.setOnClickListener(v -> new CustomDatePicker().openDialog(
+                this,
+                () -> viewModel.datePickerOpened(true),
+                (date) -> {
+                    dateField.setText((date.get(1) + 1) + "/" + date.get(2) + "/" + date.get(0));
+                    viewModel.datePickerOpened(false);
+                },
+                () -> viewModel.datePickerOpened(false)
+        ));
+
+
+        Objects.requireNonNull(binding.backArrow).setOnClickListener(v ->{} /* TODO FA-117 Navigate to Login */);
+
+        binding.createButton.setOnClickListener(v -> {
+            Objects.requireNonNull(binding.fieldsError).setVisibility(View.GONE);
+            firstNameField.clearFocus();
+            firstNameField.setSelected(false);
+            lastNameField.clearFocus();
+            lastNameField.setSelected(false);
+            emailField.clearFocus();
+            emailField.setSelected(false);
+            dateField.clearFocus();
+            dateField.setSelected(false);
+            locationField.clearFocus();
+            locationField.setSelected(false);
+            final String first = Objects.requireNonNull(firstNameField.getText()).toString();
+            final String last = Objects.requireNonNull(lastNameField.getText()).toString();
+            final String email = Objects.requireNonNull(emailField.getText()).toString();
+            final String date = Objects.requireNonNull(dateField.getText()).toString();
+            final String location = Objects.requireNonNull(locationField.getText()).toString();
+            if (!first.isEmpty() && !last.isEmpty() && !email.isEmpty() && !date.isEmpty() && !location.isEmpty()) {
+                viewModel.createAccountClicked(first, last, email, date, location);
+            } else {
+                binding.fieldsError.setVisibility(View.VISIBLE);
+            }
+        });
+
+        disposable = viewModel.getState()
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(it -> {
+                    CreateAccountUiStates state = it.getState();
+                    if (state == SUCCESS) {
```

Comment:
The UI state and the network loading state are too tied together here. The state of the Viewmodel should be of the UI, so sure, it should show loading here while it's loading. But your network call success and duplicate error messages (of which it looks like this success block has a duplicate error message too) should be separated out

Reply:
Either help me understand or move things around

Reply:
Reworked this, let me know if that is acceptable